front/src/modules/towedRollingStock/hooks/useFilterTowedRollingStock.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+
+import { osrdEditoastApi, type TowedRollingStock } from 'common/api/osrdEditoastApi';
+import { setFailure } from 'reducers/main';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+
+export interface TowedRollingStockFilters {
+  id?: number;
+  text: string;
+}
+
+const initialFilters: TowedRollingStockFilters = {
+  text: '',
+};
+
+function filterTowedRollingStocks(
+  towedRollingStockList: TowedRollingStock[],
+  filters: TowedRollingStockFilters
+) {
+  if (filters === initialFilters) return towedRollingStockList;
+  return towedRollingStockList.filter(({ id, name }) => {
+    if (filters.id !== undefined) {
+      return id === filters.id;
+    }
+    if (!filters.text) {
+      return true;
+    }
+    return name.trim().toLowerCase().includes(filters.text.trim().toLowerCase());
```

Comment:
The first `if` handles the `id` filter, and the second `if` handles the `not text` filter... then the final line handle the `text` filter. It's not really logical and also, should they be composed together? Because what if both `filter.id` and `filter.text` are defined, I believe only `filter.id` will be taken into account.

Reply:
But if you have an ID (which is unique), there is no need to match the search query.

Reply:
True, so id filtering is an early return condition, got it.